example/src/tests.ts

```diff
@@ -63,3 +65,55 @@ test("canMessage", async () => {
   const canMessage = await bob.canMessage(alice.address);
   return canMessage;
 });
+
+test("can register, encode, and decode a number codec", async () => {
+  const numberCodec = new NumberCodec();
+
+  const registry = new CodecRegistry();
+  registry.register(numberCodec);
+
+  const id = numberCodec.contentType.id();
+  const codec = registry.find(id);
+
+  const encodedContent = codec.encode(3.14);
+  const decodedContent = codec.decode(encodedContent);
+
+  return decodedContent === 3.14;
+});
+
+test("throws an error if codec is not found in registry", async () => {
+  const numberCodec = new NumberCodec();
+  const registry = new CodecRegistry();
+  registry.register(numberCodec);
+
+  try {
+    const id = "invalidId";
+    registry.find(id);
+  } catch (e) {
+    return (e as CodecError).message === "codecNotFound";
+  }
+  return false;
+});
+
+test("throws an error if codec is invalid when decoding", async () => {
+  const numberCodec = new NumberCodec();
+  const registry = new CodecRegistry();
+  registry.register(numberCodec);
+
+  try {
+    const id = numberCodec.contentType.id();
+    const codec = registry.find(id);
+
+    const encodedContent = codec.encode(3.14);
+    const invalidContentToDecode = {
+      ...encodedContent,
+      // @ts-ignore
```

Comment:
What are we ts-ignoring here?

Reply:
This `ts-ignore` is needed because this test is specifically testing the error state for an invalid codec, which clashes with the types we expect here.